docs/building-apps/network-features/vetkeys/timelock-encryption.mdx

```diff
@@ -0,0 +1,101 @@
+
+### "Dead man's switch"
+
+Journalists or whistleblowers could ensure that compromising information in their possession is automatically published if they were to become incapacitated. They can store the information in a dapp, encrypted using IBE that the dapp automatically and publicly recovers when a certain amount of time passes after it last received an authenticated ping from its owner.
+
+### Advanced timelocks
+
+Anyone can IBE-encrypt their message using a more complex condition as the identity, while a dapp allows anyone who can satisfy that condition, or when the condition is automatically fulfilled, to obtain the corresponding vetKey. This enables interesting use cases by allowing encryption tied to verifiable future events. For example, one could encrypt a message that becomes decryptable only when the price of a stock rises above or falls below a certain level, or implement information escrow and break-the-glass policies.
+
+## How to use timelock encryption?
+
+We can categorize timelock encryption into two categories: 1. scenario where the canister decrypts, and 2. scenario where the user decrypts.
+The latter is essentially the same as the [IBE](/docs/building-apps/network-features/vetkeys/identity-based-encryption), only with additional conditions on when the key can be obtained by the authorized users.
+This can be as simple as checking if enough time has passed and either returning the key or an error to the user.
+Therefore, let's focus on getting the first case to work.
+
+The code in the descriptions below is taken from the [secret-bid auction example](https://github.com/dfinity/vetkeys/tree/3b1de5d546dcc32daaf22b21baaa2b220bc857f3/examples/basic_timelock_ibe), where any user can create an auction lot with a deadline and other users can create IBE-encrypted bids for the open lots.
+The encrypted bids are decrypted by the canister once the deadline has passed and the lot was closed.
+From that moment on, the bids become public and the canister announces the winner of the lot.
+
+### Step 1: Obtain IBE public key
```

Comment:
Similar to the comments for IBE, I think it would be nice to explain the choice of parameters, like context and input. 

We could also explain why canister_id is set to none. But I don't know if it complicates more than helping